### PR TITLE
Add transformable element and `headerTitleScaleFactor`

### PR DESCRIPTION
**Pull request by blasten**
for https://github.com/PolymerElements/paper-scroll-header-panel/issues/18

### CODE REVIEW

File: paper-scroll-header-panel.html [hunk 1]
```diff
@@ -135,17 +136,28 @@
     #condensedHeaderBg {
       opacity: 0;
     }
+
+    ::content .header-title {
```
Comment (frankiefu) [inline paper-scroll-header-panel.html]: need to select the parent of the `<content>` element for compatibility with Shady DOM. e.g.

```
#headerContainer ::content .header-title {
```

File: paper-scroll-header-panel.html [hunk 2]
```diff
@@ -178,6 +190,23 @@
     properties: {
 
       /**
+       * The selector of title element in thw header.
+       */
+      headerTitleSelector: {
+        type: String,
+        value: '.title',
+        observer: '_prepareHeaderTitle'
+      },
```
Comment (frankiefu) [inline paper-scroll-header-panel.html]: I think we don't need to make the title element customizable.

File: paper-scroll-header-panel.html [hunk 3]
```diff
@@ -384,6 +427,25 @@
...
+          headerTitleElement.classList.add('header-title');
+        }
+      }
+      if (oldHeaderTitle) {
+        if (headerTitleElement = this._querySelectorInHeader(oldHeaderTitle)) {
+          headerTitleElement.classList.remove('header-title');
+        }
+      }
+    },
+
```
Comment (frankiefu) [inline paper-scroll-header-panel.html:453]: We can remove `_prepareHeaderTitle` if we don't make the `headerTitleSelector` customizable.

**Comment by blasten** [comment]
PTAL @frankiefu

**Pull request closed without merging**